tools/cmd/prepare_prebuilt_workers/main.go

```diff
@@ -137,26 +137,32 @@ func main() {
 
 			// build image
 			log.Println(fmt.Sprintf("building %s image", lang))
-			buildDockerImage := exec.Command("docker", "build", dockerfileLocation, "-t", image, "--build-arg", fmt.Sprintf("GITREF=%s", gitRef), "--build-arg", fmt.Sprintf("BREAK_CACHE=%s", uniqueCacheBreaker))
+			// TODO(jtattermusch): allow setting REPOSITORY arg to allow building from user forks e.g. https://github.com/USER_NAME/grpc.git
+			buildCommandTimeoutSeconds := 30 * 60 // 30 mins should be enough for all languages
+			buildDockerImage := exec.Command("timeout", fmt.Sprintf("%ds", buildCommandTimeoutSeconds), "docker", "build", dockerfileLocation, "-t", image, "--build-arg", fmt.Sprintf("GITREF=%s", gitRef), "--build-arg", fmt.Sprintf("BREAK_CACHE=%s", uniqueCacheBreaker))
 			buildOutput, err := buildDockerImage.CombinedOutput()
 			if err != nil {
-				log.Println(err)
-				log.Fatalf("Failed building %s image: %s", lang, string(buildOutput))
+				log.Printf("Failed building %s image. Dump of command's output will follow:\n", lang)
+				log.Println(string(buildOutput))
+				log.Fatalf("Failed building %s image: %s", lang, err.Error())
 			}
+			log.Printf("Succeeded building %s image. Dump of command's output will follow:\n", lang)
 			log.Println(string(buildOutput))
```

Comment:
In here we just dump all the buildOutput if it was built successfully , it is pretty lengthy, should we only print these build out put when an error occurred?

Reply:
Yes, that would work, we actually do something similar in our test scripts in grpc/grpc.
But it has downsides as well - sometimes is just useful to be able to inspect the log for successful builds.

I think the best solution is to make sure that the logs are stored in .log files and that there's a corresponding sponge_log.xml  file. If done correctly, one could see the structured results in the resultstore UI, and be able to see the full "target log" in the resultstore UI (but the main log wouldn't be polluted by the output).
But this solution basically requires building a "task runner" method that does all this for you. In grpc/grpc we have something like this written in python, so I don't know if it makes sense to recreate it in go.